### ISSUE TITLE
Plans for Keen UI v2: Seeking contributors!

### ISSUE THREAD

**Issue by JosephusPaye** [JosephusPaye/Keen-UI#483]
Over the next few months I'll be working more on Keen UI, and I'm looking for help to get the following done by early next year.

The things in this list are just a general indication of where I want to go, and may change at any time.

# Plans

## Immediate plans

No breaking changes, just fixes and preparation for next major version. Aiming for Christmas.

- [ ] Switch build system to Vue CLI
    - [ ] Add Prettier
    - [ ] Adapt Vue's recommended code style
- [ ] Automate the release process to enable faster, more regular updates
    - [ ] Single command to build, release to Github and npm, and update website
    - [ ] Automated change logs and release notes
- [ ] Rewrite website (no redesign yet) and docs to use Markdown and VuePress
- [ ] Add Storybook for component development
- [ ] Fix or address outstanding issues and PRs

## Further plans

Aiming for February/March.

- TypeScript declarations
- First class CSS API and theming support: make it easy to restyle and customize the appearance of all components
    - Document CSS API (e.g. major classes) for each component. I quite like the [Reach UI approach](https://reacttraining.com/reach-ui/styling).
    - Ship three themes:
        - bare (bare minimal styles (mostly layout), meant as a base for customization and other themes)
        - light (default material theme, builds bare)
        - dark (dark material theme, builds on bare)
- Create static design assets for each component (for Figma, Sketch)
- Re-examine overall component API and how everything fits together. The [Reach UI](https://github.com/reach/reach-ui) APIs are quite sensible. Can learn a thing or two from them.
    - Move away from refs + methods where possible
    - Use context to simplify coupled components like `UiTabs` + `UiTab`, `UiSelect` + `UiSelectOption`
    - Fix components that have composition issues: e.g. `UiSlider` in `UiModal`
- SSR Support

## Going forward

- Rewrite components with TypeScript and take advantage of the new features in Vue 3. This **will not** affect users of Keen UI (unless they're using TypeScript)
- Automated tests for components that need it
- Remove Sass in favor of new CSS API
- Create a Keen UI plugin for Vue CLI

# Can you help?

To get all this done (and more going forward) I'm seeking contributors to help with issues, pull requests, documentation, and code.

Please respond to this thread with how you can help if you're interested.

**Comment by kilobyte2007** [comment]
Hey @JosephusPaye,
I'd be willing to help out with some of the above. We use keen-ui in our app and I would be happy to help move the project further. Where could I be of help?

**Comment by JosephusPaye** [comment]
At the moment I could use some help writing stories for the components.

I've started this work on the [storybook branch](https://github.com/JosephusPaye/Keen-UI/tree/storybook) with stories already written for these components:

- [UiAlert](https://github.com/JosephusPaye/Keen-UI/tree/storybook/stories/UiAlert)
- [UiAutocomplete](https://github.com/JosephusPaye/Keen-UI/tree/storybook/stories/UiAutocomplete)
- [UiButton](https://github.com/JosephusPaye/Keen-UI/tree/storybook/stories/UiButton)
- [UiPopover](https://github.com/JosephusPaye/Keen-UI/tree/storybook/stories/UiPopover)
- [UiRippleInk](https://github.com/JosephusPaye/Keen-UI/tree/storybook/stories/UiRippleInk)
- [UiTooltip](https://github.com/JosephusPaye/Keen-UI/tree/storybook/stories/UiTooltip)

You could make a PR to the `storybook` branch add stories for any of the remaining components (one PR per component). Currently I'm just converting the examples in the docs to Storybook stories.

Let me know if you have any questions.

**Comment by kilobyte2007** [comment]
One question. How will be the components API represented using Storybook? In current docs we have the API block for every component, showing every component's props, methods, slots, and events, so after looking through the components you already implemented storybooks for, I could not find how this will be done.

**Comment by JosephusPaye** [comment]
They'll remain in the docs for now - which will also have the single "interactive" example.

All other examples and actual component development will happen in Storybook.

**Comment by dsimon-onventis** [comment]
[Material Design's text fields](https://material.io/components/text-fields) changed their appearance. Do you plan to adapt to those changes?

**Comment by JosephusPaye** [comment]
Yes, those will be good to have, as well as the other new changes.
